### DISCUSSION TITLE
AtTiny3227 programing issue

### DISCUSSION THREAD

**Discussion by francuz152** [SpenceKonde/megaTinyCore#547] [Q&A] [answered]
hej, when trying to program a 3227 using a uno as jtag2updi i get an error (pasted below)
everything work fine for most other attinys i tryed including 1627

it look like the 3227 is not added to the programer

```
Arduino:1.8.16 (Windows 10), Płytka:"ATtiny3227/3217/1627/1617/1607/827/817/807/427/417, ATtiny3227, 20 MHz internal, 1.8V (5 MHz or less), Disabled/Disabled, EEPROM retained, Enabled (default timer), UPDI (no reset pin), 8ms, Ignore (saves flash, almost always fine)"


Szkic używa 19937 bajtów (60%) pamięci programu. Maksimum to 32768 bajtów.

Zmienne globalne używają 981 bajtów (31%) pamięci dynamicznej, pozostawiając 2091 bajtów dla zmiennych lokalnych. Maksimum to 3072 bajtów.

avrdude: AVR Part "attiny3227" not found.



Valid parts are:

  uc3a0512 = AT32UC3A0512

  c128     = AT90CAN128

  c32      = AT90CAN32

  c64      = AT90CAN64

  pwm2     = AT90PWM2

  pwm216   = AT90PWM216

  pwm2b    = AT90PWM2B

  pwm3     = AT90PWM3

  pwm316   = AT90PWM316

  pwm3b    = AT90PWM3B

  1200     = AT90S1200

  2313     = AT90S2313

  2333     = AT90S2333

  2343     = AT90S2343

  4414     = AT90S4414

  4433     = AT90S4433

  4434     = AT90S4434

  8515     = AT90S8515

Podczas ładowania szkicu wystąpił błąd

  8535     = AT90S8535

  usb1286  = AT90USB1286

  usb1287  = AT90USB1287

  usb162   = AT90USB162

  usb646   = AT90USB646

  usb647   = AT90USB647

  usb82    = AT90USB82

  m103     = ATmega103

  m128     = ATmega128

  m1280    = ATmega1280

  m1281    = ATmega1281

  m1284    = ATmega1284

  m1284p   = ATmega1284P

  m1284rfr2 = ATmega1284RFR2

  m128rfa1 = ATmega128RFA1

  m128rfr2 = ATmega128RFR2

  m16      = ATmega16

  m1608    = ATmega1608

  m1609    = ATmega1609

  m161     = ATmega161

  m162     = ATmega162

  m163     = ATmega163

  m164p    = ATmega164P

  m168     = ATmega168

  m168p    = ATmega168P

  m168pb   = ATmega168PB

  m169     = ATmega169

  m16u2    = ATmega16U2

  m2560    = ATmega2560

  m2561    = ATmega2561

  m2564rfr2 = ATmega2564RFR2

  m256rfr2 = ATmega256RFR2

  m32      = ATmega32

  m3208    = ATmega3208

  m3209    = ATmega3209

  m324p    = ATmega324P

  m324pa   = ATmega324PA

  m325     = ATmega325

  m3250    = ATmega3250

  m328     = ATmega328

  m328p    = ATmega328P

  m328pb   = ATmega328PB

  m329     = ATmega329

  m3290    = ATmega3290

  m3290p   = ATmega3290P

  m329p    = ATmega329P

  m32m1    = ATmega32M1

  m32u2    = ATmega32U2

  m32u4    = ATmega32U4

  m406     = ATMEGA406

  m48      = ATmega48

  m4808    = ATmega4808

  m4809    = ATmega4809

  m48p     = ATmega48P

  m48pb    = ATmega48PB

  m64      = ATmega64

  m640     = ATmega640

  m644     = ATmega644

  m644p    = ATmega644P

  m644rfr2 = ATmega644RFR2

  m645     = ATmega645

  m6450    = ATmega6450

  m649     = ATmega649

  m6490    = ATmega6490

  m64m1    = ATmega64M1

  m64rfr2  = ATmega64RFR2

  m8       = ATmega8

  m808     = ATmega808

  m809     = ATmega809

  m8515    = ATmega8515

  m8535    = ATmega8535

  m88      = ATmega88

  m88p     = ATmega88P

  m88pb    = ATmega88PB

  m8u2     = ATmega8U2

  t10      = ATtiny10

  t11      = ATtiny11

  t12      = ATtiny12

  t13      = ATtiny13

  t15      = ATtiny15

  t1604    = ATtiny1604

  t1606    = ATtiny1606

  t1607    = ATtiny1607

  t1614    = ATtiny1614

  t1616    = ATtiny1616

  t1617    = ATtiny1617

  t1624    = ATtiny1624

  t1626    = ATtiny1626

  t1627    = ATtiny1627

  t1634    = ATtiny1634

  t20      = ATtiny20

  t202     = ATtiny202

  t204     = ATtiny204

  t212     = ATtiny212

  t214     = ATtiny214

  t2313    = ATtiny2313

  t24      = ATtiny24

  t25      = ATtiny25

  t26      = ATtiny26

  t261     = ATtiny261

  t28      = ATtiny28

  t3216    = ATtiny3216

  t3217    = ATtiny3217

  t4       = ATtiny4

  t40      = ATtiny40

  t402     = ATtiny402

  t402w    = ATtiny402w

  t404     = ATtiny404

  t406     = ATtiny406

  t412     = ATtiny412

  t414     = ATtiny414

  t416     = ATtiny416

  t417     = ATtiny417

  t424     = ATtiny424

  t426     = ATtiny426

  t427     = ATtiny427

  t4313    = ATtiny4313

  t43u     = ATtiny43u

  t44      = ATtiny44

  t441     = ATtiny441

  t45      = ATtiny45

  t461     = ATtiny461

  t5       = ATtiny5

  t804     = ATtiny804

  t806     = ATtiny806

  t807     = ATtiny807

  t814     = ATtiny814

  t816     = ATtiny816

  t817     = ATtiny817

  t824     = ATtiny824

  t826     = ATtiny826

  t827     = ATtiny827

  t84      = ATtiny84

  t841     = ATtiny841

  t85      = ATtiny85

  t861     = ATtiny861

  t88      = ATtiny88

  t9       = ATtiny9

  x128a1   = ATxmega128A1

  x128a1d  = ATxmega128A1revD

  x128a1u  = ATxmega128A1U

  x128a3   = ATxmega128A3

  x128a3u  = ATxmega128A3U

  x128a4   = ATxmega128A4

  x128a4u  = ATxmega128A4U

  x128b1   = ATxmega128B1

  x128b3   = ATxmega128B3

  x128c3   = ATxmega128C3

  x128d3   = ATxmega128D3

  x128d4   = ATxmega128D4

  x16a4    = ATxmega16A4

  x16a4u   = ATxmega16A4U

  x16c4    = ATxmega16C4

  x16d4    = ATxmega16D4

  x16e5    = ATxmega16E5

  x192a1   = ATxmega192A1

  x192a3   = ATxmega192A3

  x192a3u  = ATxmega192A3U

  x192c3   = ATxmega192C3

  x192d3   = ATxmega192D3

  x256a1   = ATxmega256A1

  x256a3   = ATxmega256A3

  x256a3b  = ATxmega256A3B

  x256a3bu = ATxmega256A3BU

  x256a3u  = ATxmega256A3U

  x256c3   = ATxmega256C3

  x256d3   = ATxmega256D3

  x32a4    = ATxmega32A4

  x32a4u   = ATxmega32A4U

  x32c4    = ATxmega32C4

  x32d4    = ATxmega32D4

  x32e5    = ATxmega32E5

  x384c3   = ATxmega384C3

  x384d3   = ATxmega384D3

  x64a1    = ATxmega64A1

  x64a1u   = ATxmega64A1U

  x64a3    = ATxmega64A3

  x64a3u   = ATxmega64A3U

  x64a4    = ATxmega64A4

  x64a4u   = ATxmega64A4U

  x64b1    = ATxmega64B1

  x64b3    = ATxmega64B3

  x64c3    = ATxmega64C3

  x64d3    = ATxmega64D3

  x64d4    = ATxmega64D4

  x8e5     = ATxmega8E5

  avr128da28 = AVR128DA28

  avr128da32 = AVR128DA32

  avr128da48 = AVR128DA48

  avr128da64 = AVR128DA64

  avr128db28 = AVR128DB28

  avr128db32 = AVR128DB32

  avr128db48 = AVR128DB48

  avr128db64 = AVR128DB64

  avr32da28 = AVR32DA28

  avr32da32 = AVR32DA32

  avr32da48 = AVR32DA48

  avr32db28 = AVR32DB28

  avr32db32 = AVR32DB32

  avr32db48 = AVR32DB48

  avr64da28 = AVR64DA28

  avr64da32 = AVR64DA32

  avr64da48 = AVR64DA48

  avr64da64 = AVR64DA64

  avr64db28 = AVR64DB28

  avr64db32 = AVR64DB32

  avr64db48 = AVR64DB48

  avr64db64 = AVR64DB64

  ucr2     = deprecated, use 'uc3a0512'
```

**Comment by SpenceKonde** [comment]
Where did you score a 3227??? Am I behind the curve?!

On Fri, Nov 12, 2021 at 12:26 PM francuz152 ***@***.***>
wrote:

> hej, when trying to program a 3227 usinmg a uno a jtag2updi i get an error
> (pasted below)
> everything work fine for most other attinys i tryed including 1627
>
> Arduino:1.8.16 (Windows 10), Płytka:"ATtiny3227/3217/1627/1617/1607/827/817/807/427/417, ATtiny3227, 20 MHz internal, 1.8V (5 MHz or less), Disabled/Disabled, EEPROM retained, Enabled (default timer), UPDI (no reset pin), 8ms, Ignore (saves flash, almost always fine)"
>
>
>
>
>
> Szkic używa 19937 bajtów (60%) pamięci programu. Maksimum to 32768 bajtów.
>
>
>
> Zmienne globalne używają 981 bajtów (31%) pamięci dynamicznej, pozostawiając 2091 bajtów dla zmiennych lokalnych. Maksimum to 3072 bajtów.
>
>
>
> avrdude: AVR Part "attiny3227" not found.
>
>
>
>
>
>
>
> Valid parts are:
>
>
>
>   uc3a0512 = AT32UC3A0512
>
>
>
>   c128     = AT90CAN128
>
>
>
>   c32      = AT90CAN32
>
>
>
>   c64      = AT90CAN64
>
>
>
>   pwm2     = AT90PWM2
>
>
>
>   pwm216   = AT90PWM216
>
>
>
>   pwm2b    = AT90PWM2B
>
>
>
>   pwm3     = AT90PWM3
>
>
>
>   pwm316   = AT90PWM316
>
>
>
>   pwm3b    = AT90PWM3B
>
>
>
>   1200     = AT90S1200
>
>
>
>   2313     = AT90S2313
>
>
>
>   2333     = AT90S2333
>
>
>
>   2343     = AT90S2343
>
>
>
>   4414     = AT90S4414
>
>
>
>   4433     = AT90S4433
>
>
>
>   4434     = AT90S4434
>
>
>
>   8515     = AT90S8515
>
>
>
> Podczas ładowania szkicu wystąpił błąd
>
>
>
>   8535     = AT90S8535
>
>
>
>   usb1286  = AT90USB1286
>
>
>
>   usb1287  = AT90USB1287
>
>
>
>   usb162   = AT90USB162
>
>
>
>   usb646   = AT90USB646
>
>
>
>   usb647   = AT90USB647
>
>
>
>   usb82    = AT90USB82
>
>
>
>   m103     = ATmega103
>
>
>
>   m128     = ATmega128
>
>
>
>   m1280    = ATmega1280
>
>
>
>   m1281    = ATmega1281
>
>
>
>   m1284    = ATmega1284
>
>
>
>   m1284p   = ATmega1284P
>
>
>
>   m1284rfr2 = ATmega1284RFR2
>
>
>
>   m128rfa1 = ATmega128RFA1
>
>
>
>   m128rfr2 = ATmega128RFR2
>
>
>
>   m16      = ATmega16
>
>
>
>   m1608    = ATmega1608
>
>
>
>   m1609    = ATmega1609
>
>
>
>   m161     = ATmega161
>
>
>
>   m162     = ATmega162
>
>
>
>   m163     = ATmega163
>
>
>
>   m164p    = ATmega164P
>
>
>
>   m168     = ATmega168
>
>
>
>   m168p    = ATmega168P
>
>
>
>   m168pb   = ATmega168PB
>
>
>
>   m169     = ATmega169
>
>
>
>   m16u2    = ATmega16U2
>
>
>
>   m2560    = ATmega2560
>
>
>
>   m2561    = ATmega2561
>
>
>
>   m2564rfr2 = ATmega2564RFR2
>
>
>
>   m256rfr2 = ATmega256RFR2
>
>
>
>   m32      = ATmega32
>
>
>
>   m3208    = ATmega3208
>
>
>
>   m3209    = ATmega3209
>
>
>
>   m324p    = ATmega324P
>
>
>
>   m324pa   = ATmega324PA
>
>
>
>   m325     = ATmega325
>
>
>
>   m3250    = ATmega3250
>
>
>
>   m328     = ATmega328
>
>
>
>   m328p    = ATmega328P
>
>
>
>   m328pb   = ATmega328PB
>
>
>
>   m329     = ATmega329
>
>
>
>   m3290    = ATmega3290
>
>
>
>   m3290p   = ATmega3290P
>
>
>
>   m329p    = ATmega329P
>
>
>
>   m32m1    = ATmega32M1
>
>
>
>   m32u2    = ATmega32U2
>
>
>
>   m32u4    = ATmega32U4
>
>
>
>   m406     = ATMEGA406
>
>
>
>   m48      = ATmega48
>
>
>
>   m4808    = ATmega4808
>
>
>
>   m4809    = ATmega4809
>
>
>
>   m48p     = ATmega48P
>
>
>
>   m48pb    = ATmega48PB
>
>
>
>   m64      = ATmega64
>
>
>
>   m640     = ATmega640
>
>
>
>   m644     = ATmega644
>
>
>
>   m644p    = ATmega644P
>
>
>
>   m644rfr2 = ATmega644RFR2
>
>
>
>   m645     = ATmega645
>
>
>
>   m6450    = ATmega6450
>
>
>
>   m649     = ATmega649
>
>
>
>   m6490    = ATmega6490
>
>
>
>   m64m1    = ATmega64M1
>
>
>
>   m64rfr2  = ATmega64RFR2
>
>
>
>   m8       = ATmega8
>
>
>
>   m808     = ATmega808
>
>
>
>   m809     = ATmega809
>
>
>
>   m8515    = ATmega8515
>
>
>
>   m8535    = ATmega8535
>
>
>
>   m88      = ATmega88
>
>
>
>   m88p     = ATmega88P
>
>
>
>   m88pb    = ATmega88PB
>
>
>
>   m8u2     = ATmega8U2
>
>
>
>   t10      = ATtiny10
>
>
>
>   t11      = ATtiny11
>
>
>
>   t12      = ATtiny12
>
>
>
>   t13      = ATtiny13
>
>
>
>   t15      = ATtiny15
>
>
>
>   t1604    = ATtiny1604
>
>
>
>   t1606    = ATtiny1606
>
>
>
>   t1607    = ATtiny1607
>
>
>
>   t1614    = ATtiny1614
>
>
>
>   t1616    = ATtiny1616
>
>
>
>   t1617    = ATtiny1617
>
>
>
>   t1624    = ATtiny1624
>
>
>
>   t1626    = ATtiny1626
>
>
>
>   t1627    = ATtiny1627
>
>
>
>   t1634    = ATtiny1634
>
>
>
>   t20      = ATtiny20
>
>
>
>   t202     = ATtiny202
>
>
>
>   t204     = ATtiny204
>
>
>
>   t212     = ATtiny212
>
>
>
>   t214     = ATtiny214
>
>
>
>   t2313    = ATtiny2313
>
>
>
>   t24      = ATtiny24
>
>
>
>   t25      = ATtiny25
>
>
>
>   t26      = ATtiny26
>
>
>
>   t261     = ATtiny261
>
>
>
>   t28      = ATtiny28
>
>
>
>   t3216    = ATtiny3216
>
>
>
>   t3217    = ATtiny3217
>
>
>
>   t4       = ATtiny4
>
>
>
>   t40      = ATtiny40
>
>
>
>   t402     = ATtiny402
>
>
>
>   t402w    = ATtiny402w
>
>
>
>   t404     = ATtiny404
>
>
>
>   t406     = ATtiny406
>
>
>
>   t412     = ATtiny412
>
>
>
>   t414     = ATtiny414
>
>
>
>   t416     = ATtiny416
>
>
>
>   t417     = ATtiny417
>
>
>
>   t424     = ATtiny424
>
>
>
>   t426     = ATtiny426
>
>
>
>   t427     = ATtiny427
>
>
>
>   t4313    = ATtiny4313
>
>
>
>   t43u     = ATtiny43u
>
>
>
>   t44      = ATtiny44
>
>
>
>   t441     = ATtiny441
>
>
>
>   t45      = ATtiny45
>
>
>
>   t461     = ATtiny461
>
>
>
>   t5       = ATtiny5
>
>
>
>   t804     = ATtiny804
>
>
>
>   t806     = ATtiny806
>
>
>
>   t807     = ATtiny807
>
>
>
>   t814     = ATtiny814
>
>
>
>   t816     = ATtiny816
>
>
>
>   t817     = ATtiny817
>
>
>
>   t824     = ATtiny824
>
>
>
>   t826     = ATtiny826
>
>
>
>   t827     = ATtiny827
>
>
>
>   t84      = ATtiny84
>
>
>
>   t841     = ATtiny841
>
>
>
>   t85      = ATtiny85
>
>
>
>   t861     = ATtiny861
>
>
>
>   t88      = ATtiny88
>
>
>
>   t9       = ATtiny9
>
>
>
>   x128a1   = ATxmega128A1
>
>
>
>   x128a1d  = ATxmega128A1revD
>
>
>
>   x128a1u  = ATxmega128A1U
>
>
>
>   x128a3   = ATxmega128A3
>
>
>
>   x128a3u  = ATxmega128A3U
>
>
>
>   x128a4   = ATxmega128A4
>
>
>
>   x128a4u  = ATxmega128A4U
>
>
>
>   x128b1   = ATxmega128B1
>
>
>
>   x128b3   = ATxmega128B3
>
>
>
>   x128c3   = ATxmega128C3
>
>
>
>   x128d3   = ATxmega128D3
>
>
>
>   x128d4   = ATxmega128D4
>
>
>
>   x16a4    = ATxmega16A4
>
>
>
>   x16a4u   = ATxmega16A4U
>
>
>
>   x16c4    = ATxmega16C4
>
>
>
>   x16d4    = ATxmega16D4
>
>
>
>   x16e5    = ATxmega16E5
>
>
>
>   x192a1   = ATxmega192A1
>
>
>
>   x192a3   = ATxmega192A3
>
>
>
>   x192a3u  = ATxmega192A3U
>
>
>
>   x192c3   = ATxmega192C3
>
>
>
>   x192d3   = ATxmega192D3
>
>
>
>   x256a1   = ATxmega256A1
>
>
>
>   x256a3   = ATxmega256A3
>
>
>
>   x256a3b  = ATxmega256A3B
>
>
>
>   x256a3bu = ATxmega256A3BU
>
>
>
>   x256a3u  = ATxmega256A3U
>
>
>
>   x256c3   = ATxmega256C3
>
>
>
>   x256d3   = ATxmega256D3
>
>
>
>   x32a4    = ATxmega32A4
>
>
>
>   x32a4u   = ATxmega32A4U
>
>
>
>   x32c4    = ATxmega32C4
>
>
>
>   x32d4    = ATxmega32D4
>
>
>
>   x32e5    = ATxmega32E5
>
>
>
>   x384c3   = ATxmega384C3
>
>
>
>   x384d3   = ATxmega384D3
>
>
>
>   x64a1    = ATxmega64A1
>
>
>
>   x64a1u   = ATxmega64A1U
>
>
>
>   x64a3    = ATxmega64A3
>
>
>
>   x64a3u   = ATxmega64A3U
>
>
>
>   x64a4    = ATxmega64A4
>
>
>
>   x64a4u   = ATxmega64A4U
>
>
>
>   x64b1    = ATxmega64B1
>
>
>
>   x64b3    = ATxmega64B3
>
>
>
>   x64c3    = ATxmega64C3
>
>
>
>   x64d3    = ATxmega64D3
>
>
>
>   x64d4    = ATxmega64D4
>
>
>
>   x8e5     = ATxmega8E5
>
>
>
>   avr128da28 = AVR128DA28
>
>
>
>   avr128da32 = AVR128DA32
>
>
>
>   avr128da48 = AVR128DA48
>
>
>
>   avr128da64 = AVR128DA64
>
>
>
>   avr128db28 = AVR128DB28
>
>
>
>   avr128db32 = AVR128DB32
>
>
>
>   avr128db48 = AVR128DB48
>
>
>
>   avr128db64 = AVR128DB64
>
>
>
>   avr32da28 = AVR32DA28
>
>
>
>   avr32da32 = AVR32DA32
>
>
>
>   avr32da48 = AVR32DA48
>
>
>
>   avr32db28 = AVR32DB28
>
>
>
>   avr32db32 = AVR32DB32
>
>
>
>   avr32db48 = AVR32DB48
>
>
>
>   avr64da28 = AVR64DA28
>
>
>
>   avr64da32 = AVR64DA32
>
>
>
>   avr64da48 = AVR64DA48
>
>
>
>   avr64da64 = AVR64DA64
>
>
>
>   avr64db28 = AVR64DB28
>
>
>
>   avr64db32 = AVR64DB32
>
>
>
>   avr64db48 = AVR64DB48
>
>
>
>   avr64db64 = AVR64DB64
>
>
>
>   ucr2     = deprecated, use 'uc3a0512'
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/megaTinyCore/discussions/547>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW3LNHJPRTX4NPCJGILULVE4RANCNFSM5H5KHSJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***

**Reply by francuz152** [reply]
https://www.tme.eu/pl/details/attiny3227-mu/rodzina-avr-8-bit/microchip-technology/

**Comment by SpenceKonde** [comment]
This should be fixed in the avrdude.conf in github.

**Reply by francuz152** [reply]
can you point me to the specific link? the only avrdude.conf i manage to find was updated 9 months ago

**Comment by SpenceKonde** [comment]
That would be because I hadn't actually pushed it to the repo.

**Comment by SpenceKonde** [comment, marked as answer]
https://github.com/SpenceKonde/megaTinyCore/blob/master/megaavr/avrdude.conf

**Reply by francuz152** [reply]
everything work fine now. than you
